automation/pyproject.toml

```diff
@@ -0,0 +1,24 @@
+[build-system]
+requires = ["setuptools >= 67.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "briefcase-automation"
+version = "0.1.0"
+description = "A Briefcase plugin for CI automation."
+license = {text = "BSD-3-Clause"}
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "License :: OSI Approved :: New BSD",
+    "Operating System :: OS Independent",
+]
+dependencies = [
+    "briefcase >= 0.3.17.dev100",
```

Comment:
Ok....this took me a minute. Your hyperlink returns a 404; so, I was frantically searching the internet for what on earth this is....and then I realized this all some brand new stuff for toga 🤣 Anyway, now that I understand, I got it working.

Reply:
hmm....my first attempt at this is hampered by GitHub's PR/CI tactics.

The version of Briefcase [installed](https://github.com/beeware/briefcase/actions/runs/6985085377/job/19008965506#step:9:84) in to CI is `briefcase-0.3.17.dev118+g169210e9`.

The version of Briefcase wanted by the Automation plugin is `briefcase==0.3.17.dev117+g10ed6d70`.

I'm pretty certain this is stemming from GitHub creating a new merge commit for CI for the PR and that's what `actions/checkout` is fetching....therefore, the Briefcase artifact is one commit ahead.

Reply:
hmm....this is non-trivial...but not impossible, of course. If we're going to put some kind of constraint on the version of Automation matching the version of Briefcase, then the two will always need to be sourced from the same place.

Given that Briefcase can be installed several different ways in to CI, each method will need to accommodate also installing Automation...or packaging Automation as well so it can be installed later.

Probably won't have time tonight but I'll return to this.

Reply:
Apologies for the run-around on this - for posterity, the link should have been to https://github.com/beeware/setuptools_dynamic_dependencies.

As for the version mismatch - I'm intrigued that this doesn't happen in Toga... I'm guessing the reason is that we're explicitly building wheels and installing those wheel, rather than installing from the repo directly.

Reply:
### Scenarios for how Briefcase is installed
- CI event for beeware/briefcase PR
  - Briefcase is packaged from CI's PR merge commit and wheel is installed
  - Automation should be installed from CI's PR merge commit
- CI event for beeware/briefcase that is not a PR
  - Briefcase is packaged for the current git ref and wheel is installed
    - The most common situation for this is CI running after a PR is merged in to `main`
    - Although, the git ref could also be a tag such as when releases are cut
  - Automation should be installed from the same git ref
- CI event for other repo
  - Briefcase is installed from a git ref relevant for the event
  - Automation should be installed for the same git ref

### Strategy
So, while it is relatively straightforward to say "use the same git ref to install Automation"...the ref used to install Briefcase may not be readily available arbitrarily later in the CI run. So, it'll probably be easiest to package and/or install Automation at the points in the CI when Briefcase is packaged and installed.

Reply:
This strategy turned out not to be too invasive and actually pretty straightforward. One [complication](https://github.com/beeware/.github/pull/69/files#r1406966084) from running `app-build-verify` for PRs for versioned branches of template repos where the automation plugin will not be available (or supported).